packages/now-build-utils/test/unit.builds-and-routes-detector.test.ts

```diff
@@ -452,6 +452,18 @@ describe('Test `detectBuilders`', () => {
     expect(errors![0].code).toBe('invalid_function_runtime');
   });
 
+  it('no missing version on special npm import', async () => {
+    const functions = { 'api/index.pc': { runtime: 'https://haha' } };
```

Comment:
The [`builds`](https://zeit.co/docs/configuration#project/builds) property supported anything that npm install did but I'm not sure if we want to add this feature to [`functions`](https://zeit.co/docs/configuration#project/functions) as well. There is no way to enforce a version number for any arbitrary URL for example.

Reply:
Well allowing just npm registry imports is very annoying for users that are building custom runtimes. There is no way to test your builder without pushing to npm. This also makes it very hard to do CI for custom runtimes as you cant just specify a git url + tag to import from.

Also I think this is a very arbitrary limitation - I understand that the npm import resolver is very overcomplicated, but this is just locking user choice behind an arbitrary platform limitation. This has no security impact for example as arbitrary URLs are already allowed if they are imported via the builder. Also you could have a npm 'tag' that looks like a semver version to overcome the current semver tag requirement. It might be better to remove the entire limitation of needing to have a version tag.

I also looked at the blame for this behaviour, and I think the requirement for having version tags was more of a side effect of a refactor than an intentional new behaviour. Looking at the docs for example still shows that this was previously possible - it is out of date with the current state of the code: 

```json
{
  "functions": {
    "api/test.php": {
      "runtime": "now-php"
    }
  }
}
```
(https://zeit.co/docs/configuration#project/builds)